instrumentation/opentelemetry_finch/lib/opentelemetry_finch.ex

```diff
@@ -0,0 +1,73 @@
+defmodule OpentelemetryFinch do
+  @moduledoc """
+  OpentelemetryFinch uses [telemetry](https://hexdocs.pm/telemetry/) handlers to
+  create `OpenTelemetry` spans.
+
+  ## Usage
+
+  In your application start:
+
+      def start(_type, _args) do
+        OpentelemetryFinch.setup()
+
+        # ...
+      end
+
+  """
+
+  require OpenTelemetry.Tracer
+
+  @typedoc "Setup options"
+  @type opts :: []
+
+  @doc """
+  Initializes and configures the telemetry handlers.
+  """
+  @spec setup(opts()) :: :ok
+  def setup(_opts \\ []) do
+    :telemetry.attach(
+      {__MODULE__, :request_stop},
+      [:finch, :request, :stop],
+      &__MODULE__.handle_request_stop/4,
+      %{}
+    )
+  end
+
+  @doc false
+  def handle_request_stop(_event, measurements, meta, _config) do
+    duration = measurements.duration
+    end_time = :opentelemetry.timestamp()
+    start_time = end_time - duration
+
+    status =
+      case meta.result do
+        {:ok, response} -> response.status
+        _ -> 0
+      end
+
+    attributes = %{
+      "http.scheme": meta.request.scheme,
+      "http.host": meta.request.host,
+      "http.port": meta.request.port,
+      "http.path": meta.request.path,
+      "http.method": meta.request.method,
+      "http.status": status,
```

Comment:
This should be `status_code`.

Attributes, the span name and the Status of the span need to follow the semantic conventions listed here https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md

Reply:
I have already updated the span name and span attributes
Thanks